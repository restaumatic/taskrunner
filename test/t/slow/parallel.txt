# This test that we can run two different tasks in parallel.
# Ordering is crudely approximated using `sleep`.

(
  taskrunner -n task1 sh -c '
    echo first
    sleep 2
    echo third
  '
) &

(
  taskrunner -n task2 sh -c '
    sleep 1
    echo second
  '
) &

wait

echo "tasks finished"
