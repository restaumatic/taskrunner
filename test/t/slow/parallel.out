-- output:
[task1] stdout | first
[task2] stdout | second
[task1] stdout | third
[toplevel] stdout | tasks finished
